var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -376,6 +386,60 @@ def patch(self):
             filter_file(r"GLOBAL minf", r"RANGE minf", "src/nrnoc/hh.mod")
             filter_file(r"TABLE minf", r":TABLE minf", "src/nrnoc/hh.mod")
 
+    @when("+cmake")
+    def patch(self):
+        # If channel-benchmark != disabled then add the hpc/channel-benchmark
+        # test repository and inject some extra CMake configuration to actually
+        # use it. This is an interim solution; the channel-benchmark repository
+        # should be made open source soon, at which point it can be a submodule
+        # of the NEURON repository and the CMake configuration can be committed
+        # there. There will still be an outstanding question of how to build
+        # NEURON against a **different** commit of the submodule from the one
+        # saved in the NEURON repository, which is important if you want to
+        # test a feature branch in channel-benchmark.
+        channel_benchmark_ref = self.spec.variants["channel-benchmark"].value
+        if channel_benchmark_ref == "disabled":
+            return
+        if not self.spec.satisfies("+tests"):
+            tty.warn(
+                "channel-benchmark!=disabled has no effect without +tests"
+            )
+        # Using commit= should always work, but it's sub-optimal if the
+        # variant value is actually a tag or branch name.
+        fetcher = GitFetchStrategy(
+            git="git@bbpgitlab.epfl.ch:hpc/channel-benchmark.git",
+            commit=channel_benchmark_ref,
+        )
+        # Use Spack's git wrapper to clone the repository, then rename it
+        # to sit in the right place. It seems awkward to clone it directly
+        # to the right place because Spack appends spack-src/
+        with Stage(
+            fetcher,
+            path=join_path(self.stage.source_path, "tmp-channel-benchmark"),
+        ) as stage:
+            fetcher.fetch()
+            os.rename(
+                stage.source_path,
+                join_path(
+                    self.stage.source_path, "external/tests/channel-benchmark"
+                ),
+            )
```

Comment:
this is interim solution but the fetching stuff makes me bit nervous :) 

Could you remind me how the `values=lambda x: True,` supposed to work in CI? i.e. default is `disabled` but then who specify the commit ref?

Reply:
I don't like it much either.

The idea is that `hpc/channel-benchmark` CI would specify the ref being tested in that repo, CoreNEURON CI would specify `master`.